src/sourmash/sketchcomparison.py

```diff
@@ -0,0 +1,138 @@
+"""
+Sketch Comparison Classes
+"""
+import numpy as np
+from dataclasses import dataclass
+
+from .signature import MinHash
+
+
+@dataclass
+class BaseMinHashComparison:
+    """Class for standard comparison between two MinHashes"""
+    mh1: MinHash
+    mh2: MinHash
+    ignore_abundance: bool = False # optionally ignore abundances
+
+    def check_comparison_compatibility(self):
+        # do we need this check + error? Minhash functions should already complain appropriately...
+        k1 = self.mh1.ksize
+        k2 = self.mh2.ksize
+        if k1 != k2:
+            raise TypeError(f"Error: Invalid Comparison, ksizes: {k1}, {k2}. Must compare sketches of the same ksize.")
+        self.ksize = self.mh1.ksize
+        m1 = self.mh1.moltype
+        m2= self.mh2.moltype
+        if m1 != m2:
+            raise TypeError(f"Error: Invalid Comparison, moltypes: {m1}, {m2}. Must compare sketches of the same moltype.")
+        self.moltype= self.mh1.moltype
+        # check num, scaled
+        if not any([(self.mh1.num and self.mh2.num), (self.mh1.scaled and self.mh2.scaled)]):
+            raise TypeError("Error: Both sketches must be 'num' or 'scaled'.")
+
+    def downsample_and_handle_ignore_abundance(self, cmp_num=None, cmp_scaled=None):
+        """
+        Downsample and/or flatten minhashes for comparison
+        """
+        if self.ignore_abundance:
+            self.mh1_cmp = self.mh1.flatten()
+            self.mh2_cmp = self.mh2.flatten()
+        else:
+            self.mh1_cmp = self.mh1
+            self.mh2_cmp = self.mh2
+        if cmp_scaled is not None:
+            self.mh1_cmp = self.mh1_cmp.downsample(scaled=cmp_scaled)
+            self.mh2_cmp = self.mh2_cmp.downsample(scaled=cmp_scaled)
+        elif cmp_num is not None:
```

Comment:
it should, but ofc only in places where we actually use the `NumMinHashComparison` ...maybe I should make an issue about going through and using these a bit more?

e.g. Search could be changed to build a search result immediately, rather than doing calculations and sorting/turning them into a result later (still need to sort, ofc)

Reply:
no, I think it's enough to put the link in for now! I've added a comment over there.